packages/core/src/postProcess/PostProcessEffectParameter.ts

```diff
@@ -0,0 +1,98 @@
+import { Color, MathUtil, Vector2, Vector3, Vector4 } from "@galacean/engine-math";
+import { Texture } from "../texture";
+
+type ValueType = number | Color | Vector2 | Vector3 | Vector4 | boolean | Texture;
+enum ValueTypeEnum {
+  Number,
+  Color,
+  Vector2,
+  Vector3,
+  Vector4,
+  Other
+}
+
+/**
+ * Represents a parameter of a post process effect.
+ * @remarks
+ * The parameter will be mixed to a final value and be used in post process manager.
+ */
+export class PostProcessEffectParameter<T> {
+  /**
+   * Whether the parameter is enabled.
+   */
+  enabled = true;
+
+  private readonly _valueType: ValueTypeEnum;
+  private readonly _canLerp: boolean;
+
+  private _value: ValueType;
+
+  /**
+   * The value of the parameter.
+   */
+  get value(): T {
+    return this._value as T;
+  }
+
+  set value(value: ValueType) {
+    if (this._valueType === ValueTypeEnum.Number) {
+      this._value = MathUtil.clamp(<number>value, this._min, this._max);
```

Comment:
Are you sure you want to write the behavior of the subclass into the parent class?